data-protocols/dsp/dsp-negotiation/dsp-negotiation-transform/src/main/java/org.eclipse.edc.protocol.dsp.negotiation.transform/to/JsonObjectToContractRequestMessageTransformer.java

```diff
@@ -40,28 +41,51 @@ public JsonObjectToContractRequestMessageTransformer() {
     }
 
     @Override
-    public @Nullable ContractRequestMessage transform(@NotNull JsonObject object, @NotNull TransformerContext context) {
-        var builder = ContractRequestMessage.Builder.newInstance()
-                .protocol(DATASPACE_PROTOCOL_HTTP)
-                .type(INITIAL)
-                .processId(transformString(object.get(DSPACE_NEGOTIATION_PROPERTY_PROCESS_ID), context))
-                .callbackAddress(transformString(object.get(DSPACE_NEGOTIATION_PROPERTY_CALLBACK_ADDRESS), context))
-                .dataSet(transformString(object.get(DSPACE_NEGOTIATION_PROPERTY_DATASET), context));
-
-        var policy = transformObject(object.get(DSPACE_NEGOTIATION_PROPERTY_OFFER), Policy.class, context);
-        if (policy == null) {
-            context.reportProblem("Cannot transform to ContractRequestMessage with null policy");
+    public @Nullable ContractRequestMessage transform(@NotNull JsonObject requestObject, @NotNull TransformerContext context) {
+        var builder = ContractRequestMessage.Builder.newInstance().protocol(DATASPACE_PROTOCOL_HTTP);
+        if (!transformMandatoryString(requestObject.get(DSPACE_NEGOTIATION_PROPERTY_PROCESS_ID), builder::processId, context)) {
+            context.reportProblem(format("No '%s' specified on ContractRequestMessage", DSPACE_NEGOTIATION_PROPERTY_PROCESS_ID));
             return null;
         }
 
-        var contractOffer = ContractOffer.Builder.newInstance()
-                .id(nodeId(object))
-                .assetId(policy.getTarget())
-                .policy(policy).build();
+        var callback = requestObject.get(DSPACE_NEGOTIATION_PROPERTY_CALLBACK_ADDRESS);
+        if (callback != null) {
+            transformString(callback, builder::callbackAddress, context);
+        }
 
-        builder.contractOffer(contractOffer);
+        var dataset = requestObject.get(DSPACE_NEGOTIATION_PROPERTY_DATASET);
+        if (dataset != null) {
+            builder.dataSet(transformString(dataset, context));
+        }
 
+        var contractOffer = requestObject.get(DSPACE_NEGOTIATION_PROPERTY_OFFER);
+        if (contractOffer != null) {
+            if (!(contractOffer instanceof JsonObject)) {
```

Comment:
I'm not sure that in an expanded jsonld the offer would be found in a `JsonObject`, I think it would be in a `JsonArray` similarly as stated here: https://github.com/eclipse-edc/Connector/issues/2919

the `transformObject` at line 68 will already handle this case, but this guard will be wrongly hit

maybe we need some e2e tests to verify this behavior

Reply:
OK a flagged this for investigation in the other issue